### DISCUSSION TITLE
How to print each line individually, rather than a full paragraph? (Export for Web)

### DISCUSSION THREAD

**Discussion by ahannbyrd** [inkle/ink#869] [General]
Hi all! Long-time narrative writer with amateur coding chops here, and I was hoping someone could help me out with a challenge I'm up against...

In the [RunningYourInk](https://github.com/inkle/ink/blob/master/Documentation/RunningYourInk.md) documentation, it says:

> Often when designing the flow for your game, the sequence of interactions between the player and the story may not precisely match the way the ink is evaluted. For example, with a classic choose-your-own-adventure type story, you may want to show multiple lines (paragraphs) of text and choices all at once. For a visual novel, you may want to display one line per screen.

I just can't figure out how to do that.  I used Inky's "Export for Web" feature to create an HTML folder and can't figure out what to change in Main.js so that each line of dialogue prints individually and then the player has to click to continue for each, until they receive a choice.

Assuming it has to do with this section?

> // Main story processing function. Each time this is called it generates
    // all the next content up as far as the next set of choices.
    function continueStory(firstTime) {

        var paragraphIndex = 0;
        var delay = 0.0;

        // Don't over-scroll past new content
        var previousBottomEdge = firstTime ? 0 : contentBottomEdgeY();

        // Generate story text - loop through available content
        while(story.canContinue) {

            // Get ink to generate the next paragraph
            var paragraphText = story.Continue();
            var tags = story.currentTags;

or this one?

> // Create paragraph element (initially hidden)
>             var paragraphElement = document.createElement('p');
>             paragraphElement.innerHTML = paragraphText;
>             storyContainer.appendChild(paragraphElement);
> 
>             // Add any custom classes derived from ink tags
>             for(var i=0; i<customClasses.length; i++)
>                 paragraphElement.classList.add(customClasses[i]);
> 
>             // Fade in paragraph after a short delay
>             showAfter(delay, paragraphElement);
>             delay += 200.0;

Any help would be much appreciated, thanks!

- Adam

**Comment by balthane** [comment]
I don't know if you ever figured this out, but in case others are looking for the same answer.  Instead of 

   while(story.canContinue) {

Change to
 if(story.canContinue) {

**Reply by ahannbyrd** [reply]
Thanks for this! In the end, I wound up using a work around by creating a knot in the ink file itself, then tunnelling through it any time I wanted to force a pause. Sneaky, though admittedly a bit tedious!

=== Pause ===
+ [Continue]
->->
